### ISSUE TITLE
Likely regression in track handling between versions `20240101T000844` and `20240401T222316`

### ISSUE THREAD

**Issue by maxz** [saramibreak/DiscImageCreator#275]
**Version**
`64 bit, AnsiBuild, 20240401T222316`

**Describe the bug**
The first 75 sectors of the second track for the disc mentioned below are scrambled.

DIC treats the track as an audio track and descrambles the first 75 sectors. The latest version comes to a result which differs from version `20240101T000844` and also from Redumper. Redumper treats the whole track as a data track.

Also see: https://github.com/superg/redumper/issues/148
I first wrongly reported this as an issue in Redumper.

**Disc title**
[PC Action 9/99](http://redump.org/disc/92982/)

**Log files**
[PCABONUS0999 (DiscImageCreator).zip](https://archive.org/download/redump.org-logs/PCABONUS0999%20%28DiscImageCreator%29.zip)
[PCABONUS0999 (Redumper).zip](https://archive.org/download/redump.org-logs/PCABONUS0999%20%28Redumper%29.zip)
`DIC 20240101T000844` and `Redumper build 314`: [PC Action 1999-09 (Bonus-CD).zip](https://archive.org/download/redump.org-logs/PC%20Action%201999-09%20%28Bonus-CD%29.zip)

**Comment by saramibreak** [comment]
I was misled by this report. http://forum.redump.org/topic/53657/fixed-cdi-track-2-wrong-in-cdi-video-cd-titelneuheiten-ii95

Uploaded.
Linux [DiscImageCreator_test.tar.gz](https://github.com/saramibreak/DiscImageCreator/files/15369766/DiscImageCreator_test.tar.gz)
Windows [DiscImageCreator_test.zip](https://github.com/saramibreak/DiscImageCreator/files/15369767/DiscImageCreator_test.zip)

Btw, it seems he is busy, but I'm not. Please report a bug firstly to me. Your report is always welcome, whatever it is.

**Comment by maxz** [comment]
<details><summary>Your provided test version produces the correct checksums again.</summary>

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE datafile PUBLIC "-//Logiqx//DTD ROM Management Datafile//EN" "http://www.logiqx.com/Dats/datafile.dtd">
<datafile>
    <header>
        <name>-insert name-</name>
        <description>-insert description-</description>
        <version>-insert version-</version>
        <date>-insert date-</date>
        <author>-insert author-</author>
        <homepage>-insert homepage-</homepage>
        <url>-insert url-</url>
    </header>
    <game name="-insert name-">
        <category>Games</category>
        <description>-insert description-</description>
        <rom name="PCABONUS0999 (Track 01).bin" size="472274544" crc="2dc274b8" md5="f0dfc99bfb3cda88608555e86a7922ca" sha1="41deb9d70a24321188e4236d7b33e1c6d64f4308"/>
        <rom name="PCABONUS0999 (Track 02).bin" size="25298112" crc="b3238145" md5="692d5ce49785260a510e180361739815" sha1="510d9d79cbf3532824a345e108d42091ef0822a2"/>
        <rom name="PCABONUS0999 (Track 03).bin" size="15360912" crc="d4b2e0de" md5="8a105fa7ba4f1c518aa11b23e15bcfb0" sha1="65811b97c03ef4dac89393989c035e1e1f828614"/>
        <rom name="PCABONUS0999 (Track 04).bin" size="27036240" crc="e9f65554" md5="6497583f25de2a7063942db0e716d92f" sha1="e45cccebeae7ca303eaa264e69b85e53ed1505aa"/>
        <rom name="PCABONUS0999 (Track 05).bin" size="23141328" crc="fd30f5b9" md5="0e9267062c7d4459e3c07b49570a97d9" sha1="e26d470cda9fe56198bdb4a5e6d972ff7625a9da"/>
        <rom name="PCABONUS0999 (Track 06).bin" size="30655968" crc="fbd2fb68" md5="63e4a69843a9f78cf6c90d4d64fb46be" sha1="d79cc8c215d8130a6d445edf7f7199b6a3c9a6df"/>
        <rom name="PCABONUS0999 (Track 07).bin" size="23738736" crc="d77e9731" md5="e9645193e4236f645c8bd113035b2b1a" sha1="659fb656044216de7bdf3613b09948bd3c5b098c"/>
        <rom name="PCABONUS0999 (Track 08).bin" size="28882560" crc="ee577992" md5="d403972c9c0ab6ed5e5ebe95190764a5" sha1="9a33b227b8ed6f91395b8d70abdb51073cda145c"/>
        <rom name="PCABONUS0999 (Track 09).bin" size="24051552" crc="fdbdb674" md5="d320cd8e48b8fd4c6c71fe9af3aff865" sha1="95bd7aa9e4dfd5b7e7510cd6104584031bdb5d37"/>
        <rom name="PCABONUS0999 (Track 10).bin" size="29461152" crc="2a44a71f" md5="602bce017b8ef53777290f563faab514" sha1="191133c9eaaec2812b5dede81520c462e2bbde4e"/>
        <rom name="PCABONUS0999 (Track 11).bin" size="25914336" crc="8ce02019" md5="b457e5ee75e33fb4aa0caa36ee16c226" sha1="d594c025edaac235e0eda9d4dda6a76c68026c20"/>
        <rom name="PCABONUS0999 (Track 12).bin" size="21577248" crc="2c79c959" md5="0a18b264e985733058afcd4d5d371743" sha1="384c121643261978b9056052e69fa394b9d0b64e"/>
        <rom name="PCABONUS0999 (Track 13).bin" size="30150288" crc="2e587780" md5="fc3a936a4b4cc28158b23143b55b2e3f" sha1="e5f43345df42ec35f303eb560768082072fca770"/>
    </game>
</datafile>
```
</details>

<details><summary>The log of the test version.</summary>

```
AppVersion
        64 bit, AnsiBuild, 20240519T232943
/c2 val3 was omitted. set [0]
/c2 val4 was omitted. set [0]
valid extension was omitted. -> set .bin
CurrentDirectory
        /tmp/PCABONUS0999
WorkingPath
         Argument: /tmp/PCABONUS0999/PCABONUS0999.bin
         FullPath: /tmp/PCABONUS0999/PCABONUS0999.bin
            Drive: /
        Directory: tmp/PCABONUS0999/
         Filename: PCABONUS0999
        Extension: .bin
StartTime: 2024-05-19T19:41:18+0200
Set the drive speed: 7056KB/sec
          Block Size: 4096
       Flagment Size: 4096
       All Block Num: 24635381
      Free Block Num: 2825099
 Available Block Num: 1567141
          I Node Num: 6275072
     Free I Node Num: 4749567
Available I Node Num: 4749567
      File System ID: 6749354239398096367
          Mount Flag: 1024
 Max Filename Length: 255
DiskSize of [/tmp/PCABONUS0999]
        Total:   100906520576 bytes
         Used:    89334915072 bytes
        ---------------------------
        Space:    11571605504 bytes
         => There is enough disk space for dumping
Check if subchannel pack mode is supported
[INFO] This drive supports subchannel pack mode
Checking SubRtoW (Track) 13/13
This drive (CD-R   PX-W4012A 1.07 03/22/06  09:00     ) supports [OpCode: 0xd8, SubCode: 0]
This drive (CD-R   PX-W4012A 1.07 03/22/06  09:00     ) supports [OpCode: 0xd8, SubCode: 1]
This drive (CD-R   PX-W4012A 1.07 03/22/06  09:00     ) supports [OpCode: 0xd8, SubCode: 2]
This drive (CD-R   PX-W4012A 1.07 03/22/06  09:00     ) supports [OpCode: 0xd8, SubCode: 8]
Checking Pregap sync, msf, mode (LBA)  -7843
Reading DirectoryRecord   21/  21
LBA[000000, 0000000]: [F:ReadTOCText][L:546]
        Opcode: 0x43
        ScsiStatus: 0x02 = CHECK_CONDITION
        SenseData Key-Asc-Ascq: 02-04-01 = NOT_READY - LUN_NOT_READY - BECOMING_READY
LBA[000000, 0000000]: [F:ReadCDForCheckingSubQAdr][L:1109]
        Opcode: 0xd8
        ScsiStatus: 0x02 = CHECK_CONDITION
        SenseData Key-Asc-Ascq: 06-29-00 = UNIT_ATTENTION - POWER ON, RESET, OR BUS DEVICE RESET OCCURRED
Please wait for 40000 milliseconds until the device is returned
lpCmd: d8, 00, 00, 00, 00, 00, 00, 00, 00, 02, 08, 00
dwBufSize: 5484
Checking SubQ adr (Track) 13/13
Creating (Track AA).bin & .sub (LBA)   330687/  330687

        There is not non-zero byte in the (Track AA)
Creating (Track 00).scm & .sub, (Track 01)(-LBA).scm & .sub (LBA)    -8508/  -10000
Set OpCode: 0xd8, SubCode: 8(Raw)
Checking SubQ ctl (Track) 13/13
Creating .scm (LBA) 330588/330588
No C2 errors
Copying .scm to .img
Set LBA to descramble sector 330587/330587
Descrambling data sector of img: 200796/200796
Exec ./EccEdc.out check Sub "/tmp/PCABONUS0999/PCABONUS0999.img"
Checking sectors: 330587/330587
[ERROR] Number of sector(s) where user data doesn't match the expected ECC/EDC: 75
Total errors: 75
Total warnings: 0
Creating cue and ccd (Track) 13/13
Creating bin (Track) 13/13
Hashing: PCABONUS0999.scm
Hashing: PCABONUS0999.img
Hashing: PCABONUS0999 (Track 01).bin
Hashing: PCABONUS0999 (Track 02).bin
Hashing: PCABONUS0999 (Track 03).bin
Hashing: PCABONUS0999 (Track 04).bin
Hashing: PCABONUS0999 (Track 05).bin
Hashing: PCABONUS0999 (Track 06).bin
Hashing: PCABONUS0999 (Track 07).bin
Hashing: PCABONUS0999 (Track 08).bin
Hashing: PCABONUS0999 (Track 09).bin
Hashing: PCABONUS0999 (Track 10).bin
Hashing: PCABONUS0999 (Track 11).bin
Hashing: PCABONUS0999 (Track 12).bin
Hashing: PCABONUS0999 (Track 13).bin
EndTime: 2024-05-19T19:53:55+0200
```
</details>

The tracks now match the old DIC version's and Redumper's results.

> Btw, it seems he is busy, but I'm not. Please report a bug firstly to me. Your report is always welcome, whatever it is.

Alright. I will report messages which are connected to both programs here first, if you insist. Especially anything connected to scrambled sectors.
So far I tried to determine the most likely source and then reported it where it seemed appropriate. But this makes it easier for me too.
